### PR TITLE
Feature nullness :: Bugfixes

### PR DESCRIPTION
**Pull request by T-Gro**
1. Ignore nullness applied on struct-based instances of generic typars allowing nullness. i.e. "T?" in a C# API where T can also be a struct.
2. Change CLI events declared as EventHandler? in C# look like without null in F#, because F# has null-safe access via Fsharp.Core adapters.
3. Bugfix a situation where a mutable let binding first starts with null (typechecker assigns the "supports null constraint"), and is later assigned a not null value (which does not meet "supports null").

### CODE REVIEW

File: src/Compiler/Checking/ConstraintSolver.fs [hunk 1]
```diff
@@ -1222,6 +1234,11 @@ and SolveTypeEqualsType (csenv: ConstraintSolverEnv) ndeep m2 (trace: OptionalTr
     let sty1 = stripTyEqnsA csenv.g canShortcut ty1
     let sty2 = stripTyEqnsA csenv.g canShortcut ty2
 
+    let csenv = 
+        match ty1 with
```
Comment (dsyme) [inline src/Compiler/Checking/ConstraintSolver.fs:1238]: This should match sty1 I think

Comment (T-Gro) [reply src/Compiler/Checking/ConstraintSolver.fs:1238]: sty1 is stripped, and stripping also removes the first "layers of typars" . The IsSupportsNullFlex flag was only on the topmost level, but not deeper.

Other option would be to change the stripping function to carry flags, but that felt to invasive (used in a lot more places.)

Comment (dsyme) [reply src/Compiler/Checking/ConstraintSolver.fs:1238]: I'm a little surprised you need to push the flag into csenv.

Comment (T-Gro) [reply src/Compiler/Checking/ConstraintSolver.fs:1238]: I was trying without it, but was losing it due to abovementioned stripping.

(e.g. a typar whose solution is another typar. Flag sitting on the top, but the constraint was lower in the chain)

**Pull request merged**